soc/riscv/riscv-privilege/nuclei/gd32vf103/pinctrl_soc.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2021 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * Gigadevice SoC specific helpers for pinctrl driver
+ */
```

Comment:
The file soc/arm/gigadevice/common/pinctrl_soc.h should be moved to include/drivers/pinctrl and renamed to pinctrl_gd32_soc.h, for instance. 
@gmarull ?

Reply:
It is a requirement to have a file named `pinctrl_soc.h` in the include path. An idea:

- Move this file somewhere else (maybe pinctrl drivers folder?), name it `pinctrl_soc_gd32.h`, then create a `pinctrl_soc.h` for both ARM and RISC-V, just including that file.

Reply:
I think it is better the header rename to gd32-pinctrl.h and put into  include/dt-bindings/pinctrl/.
The folder contains similar header for other pinctrl devices.
But put into the folder, the header must be able to included from dts files.
some definitions will be rest in soc/.../pinctrl_soc.h.
I will try to rewrite with this method.